sdk/python/tests/test_historical_retrieval.py

```diff
@@ -246,7 +271,12 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
 @pytest.mark.parametrize(
     "infer_event_timestamp_col", [False, True],
 )
-def test_historical_features_from_parquet_sources(infer_event_timestamp_col):
+@pytest.mark.parametrize(
+    "full_feature_names", [False, True],
```

Comment:
If we add this parametrize, then we double all end to end tests we are running on BigQuery, which could eventually amount to a lot of money. We are planning to refactor the tests to make them more modular, but for the time being is there a way to incorporate `full_feature_names` in such a way that it doesnt double our tests?

Reply:
gotcha, @Mwad22 i think you could write a new test method that can be very similar to the stuff in this test. I think if you put the test code within this test method even without parametrization, it'll still run that new part you add multiple times because this is parametrized by other things. WDYT? definitely feel free to suggest any other ideas

Reply:
So, I think it might be beneficial to still have these end to end tests that test the feature name, just to make sure that we are retrieving the correct whether the `full_feature_names` parameter is set to True or False. However, (as @mavysavydav suggested) I can place these tests in a separate test function so that those that are unrelated to (changes in) the `full_feature_names` are no longer parametrized by it. That should reduce the total number of tests. Please let me know what your thoughts are on this.

Reply:
> I can place these tests in a separate test function so that those that are unrelated to (changes in) the full_feature_names are no longer parametrized by it

I don't understand what this sentence means. What is "those"? and what is "it"?

Reply:
Oh my bad, I'll clarify. So, 'those' is in reference to the tests that do not need/involve testing the `full_feature_names` functionality. And `it` refers to the full_feature_names parameter. What I meant was I can place all tests that test `full_feature_names` functionality in a separate test function. So tests that do not need to be run with `parametrize(
    "full_feature_names", [False, True])` no longer do so.

Reply:
What bothers me is that this single method is becoming a monster. There are too many conditionals. I am open to the idea of having all these parameterizations and even loads of different combinations, but then the test itself should run really quickly.

What if we added a copy of this test that is very small. Like a `test_historical_features_from_bigquery_sources_minimal()`. We then try to ensure that it only uses like 100-1000 rows and executes all of the tests very quickly with all the combinations. I havent checked, but I believe the existing tests actually uses a lot more data. (Btw I only care about BQ. It's fine to let Parquet run over and over since its fast).

Reply:
So after doing a little digging, I believe the data being user is limited to 1000 rows by the hardcoded `order_count` in the `generate_entities` function [here](https://github.com/Mwad22/feast/blob/d59e0c3cebd49306dbe8b6006400117435acbae7/sdk/python/tests/test_historical_retrieval.py#L37). Is this sufficient or would a making the order count a variable parameter (and setting it to 100) be more ideal?

Reply:
I think it's probably fine in that case.

Reply:
Cool, I added the `order_count` parameter to the `generate_entities()` so we are able to vary the number of rows generated. It's an optional parameter with the default value of `order_count=1000`, so that this change does not interfere with any of the current code.